### ISSUE TITLE
com.google.gson.stream.MalformedJsonException for EntityStream

### ISSUE THREAD

**Issue by anyili** [redis/redis-om-spring#124]
I was following the example to write query using EntityStream. Something like 
```java
List<String> results =  entityStream
                .of(SomeDocument.class)
                .filter(SomeDocument$.NAME.eq(name))
                .limit(1000
                ).map(SomeDocument$.DESCRIPTION)
                .collect(Collectors.toList());
        results.stream().forEach(System.out::println);
```

However, I always got `com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 12 path $`
Any ideas? Thanks

**Comment by bsbodden** [comment]
Can you post the code for the `SomeDocument` or the equivalent that you're using and perhaps a simple example to create one?

**Comment by bsbodden** [comment]
@anyili can you post a reproducer with some sample data? I'm curious as to what triggers the malformed JSON. Thanks!

**Comment by chayim** [comment]
CC @sazzad16 maybe of interest to you too?

**Comment by anyili** [comment]
The `SomeDocument` class will be like the below

```java 
@Data
@NoArgsConstructor
@Document
public class SomeDocument  {
    @NonNull
    @Id
    @Indexed
    protected String id;
    @NonNull
    @TagIndexed
    private String name;

    @Indexed(sortable = true)
    private LocalDateTime documentCreationDate;
    @Searchable
    private String description;
    @TagIndexed
    private Source source;
    @Searchable
    private String category;
    @TagIndexed
    private Format format;
    @Searchable
    private String objectStorageKey;

    @TextIndexed
    private String searchableContent;

    public enum Format {
        pdf,
        word,
        text,
        png,
        jpeg
    }

    public enum Source {
      sourceA,
      sourceB
    }




    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        SomeDocument that = (SomeDocument) o;

        if (!id.equals(that.id)) return false;
        return name.equals(that.name);
    }

    @Override
    public int hashCode() {
        int result = id.hashCode();
        result = 31 * result + name.hashCode();
        return result;
    }


}
```

**Comment by bsbodden** [comment]
Confirmed and fixed, it should be out on `0.6.4` @anyili I wanted to post the proper way to annotate that class `@TagIndex` and `@TextIndexed` will be likely deprecated soon in favor of just using `@Indexable` and `@Searchable`. The example class should look like:

```
package com.redis.om.spring.annotations.document.fixtures;

import com.redis.om.spring.annotations.*;
import org.springframework.data.annotation.Id;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import java.time.LocalDateTime;

@Data @NoArgsConstructor @Document public class SomeDocument {
  @NonNull @Id @Indexed protected String id;
  @NonNull @Indexed private String name;

  @Indexed(sortable = true) private LocalDateTime documentCreationDate;
  @Searchable private String description;
  @Indexed private Source source;
  @Searchable private String category;
  @Indexed private Format format;
  @Searchable private String objectStorageKey;
  @Searchable private String searchableContent;

  public enum Format {
    pdf,
    word,
    text,
    png,
    jpeg
  }

  public enum Source {
    sourceA,
    sourceB
  }

  @Override public boolean equals(Object o) {
    if (this == o)
      return true;
    if (o == null || getClass() != o.getClass())
      return false;

    SomeDocument that = (SomeDocument) o;

    if (!id.equals(that.id))
      return false;
    return name.equals(that.name);
  }

  @Override public int hashCode() {
    int result = id.hashCode();
    result = 31 * result + name.hashCode();
    return result;
  }

}
```

Cheers,
BSB

**Comment by anyili** [comment]
@bsbodden  Great. Thanks. Will try it once it is out.
